frontend/src/components/ShareModal/hooks/useSearch.ts

```diff
@@ -0,0 +1,288 @@
+
+  const debouncedSearchInputValue = useDebounce<string>(
+    state.searchInputValue,
+    500,
+  );
+
+  const { isAdml } = useIsAdml();
+  const { appCode } = useOdeClient();
+
+  useEffect(() => {
+    search(debouncedSearchInputValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debouncedSearchInputValue]);
+
+  const handleSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    dispatch({
+      type: "onChange",
+      payload: value,
+    });
+  };
+
+  const search = async (debouncedSearchInputValue: string) => {
+    dispatch({
+      type: "isSearching",
+      payload: true,
+    });
+    // start search from 1 caracter length for non Adml but start from 3 for Adml
+    if (
+      (!isAdml && debouncedSearchInputValue.length >= 1) ||
+      (isAdml && debouncedSearchInputValue.length >= 3)
+    ) {
+      const resSearchShareSubjects = await odeServices
+        .share()
+        .searchShareSubjects(
+          appCode,
+          resource.assetId,
+          debouncedSearchInputValue,
+        );
+
+      dispatch({
+        type: "addApiResult",
+        payload: resSearchShareSubjects,
+      });
+
+      const adaptedResults = resSearchShareSubjects
+        // exclude subjects that are already in the share table
+        .filter(
+          (right: { id: any }) =>
+            !shareRights.rights.find(
+              (shareRight: { id: any }) => shareRight.id === right.id,
+            ),
+        )
+        // exclude owner from results
+        .filter(
+          (right: { type: string; id: any }) =>
+            !(right.type === "user" && right.id === resource.creatorId),
+        )
+        .map((searchResult: { id: any; displayName: any; type: string }) => {
+          return {
+            value: searchResult.id,
+            label: searchResult.displayName,
+            icon: searchResult.type === "sharebookmark" ? Bookmark : null,
+          };
+        });
+
+      dispatch({
+        type: "addResult",
+        payload: adaptedResults,
+      });
+    } else {
+      dispatch({
+        type: "emptyResult",
+        payload: [],
+      });
+      Promise.resolve();
+    }
+
+    dispatch({
+      type: "isSearching",
+      payload: false,
+    });
+  };
+
+  const handleSearchResultsChange = async (model: Array<string | number>) => {
+    const shareSubject = state.searchAPIResults.find(
+      (searchAPIResult) => searchAPIResult.id === model[0],
+    );
+
+    if (shareSubject) {
+      let rightsToAdd: ShareRight[] = [];
+
+      if (shareSubject.type === "sharebookmark") {
+        const bookmarkRes = await odeServices
+          .directory()
+          .getBookMarkById(shareSubject.id);
+
+        rightsToAdd.push({
+          ...bookmarkRes,
+          type: "sharebookmark",
+          avatarUrl: "",
+          directoryUrl: "",
+          actions: defaultActions,
+        });
+
+        bookmarkRes?.users
+          .filter(
```

Comment:
* Faire `filter().forEach()`, c'est itérer 2x sur le tableau quand 1 seule fois suffit dans le forEach => gain de perf facile.
* Un code similaire existe dans le hook `useShare`, où l'on parcourt le tableau `rights` X fois pour en extraire 3 sous-tableaux `users / groups / bookmarks` => à factoriser ?